### PR TITLE
Adding support for config JSON file and dynamic module imports

### PR DESCRIPTION
**Pull request by rennanharo**
# Context
We must be able to control which extractions to run, with different arguments for each. Also, keeping in mind that the number of extraction sources will increase, so we should have a very flexible way of doing so. 

# Proposed changes
Use a JSON file as a `config source` to control what to run and with which arguments.
To achieve that, we must be able to dynamically import and execute each one of the modules passed in on the config file.

Example config file:
```json
{
  "CNAC": {
    "source_path": "./modules/extr_cnac.py",
    "source_args": ""
  },
  "RCIV": {
    "source_path": "./modules/extr_rciv.py",
    "source_args": ["2021-01-01 00:00:00", "2021-02-28 00:00:00"]
  },
  "SRAG": {
    "source_path": "./modules/extr_srag.py",
    "source_args": ""
  }
}
```
The script should be able to dynamically import the module by its given path, and execute its' `.download()` method.

Also, I'm proposing a better file organisation given our perspective of codebase growth.
The proposed structure is:
```
.
├── extractors
│   ├── modules
│   │   ├── extr_cnac.py
│   │   ├── extr_dgbr.py
│   │   ├── extr_rciv.py
│   │   ├── extr_registro_civil.py
│   │   └── extr_srag.py
|   |
|   ├── config.json
|   ├── Dockerfile
|   ├── extract.py
│   └── requirements.txt
```

# Types of changes
- [x] Feature improvement (non-breaking change which improves an existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Hotfix (unexpected change which fixes an issue as an emergency response to an event or incident)
- [ ] Bugfix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by brunobraziel** [comment]
Achei bem interessante o que você fez pra que ficasse mais fácil de reutilizar o código. Gostaria de fazer 2 sugestões e podemos discutir sobre:

1. Se chamássemos esse aquivo main.py de extractions_central.py, uma vez que ele seria um intermediador entre todos os extratores, e seria bem mais fácil de, em um futuro, quando isso for documentado, entender o propósito do arquivo.
2. Se criássemos um arquivo JSON passando as chaves permitidas (CNAC, RCIV e SRAG), com seus valores sendo a chamada dos métodos. Dessa forma, conforme nossas extrações forem expandindo, poderíamos apenas popular este json sem ter que tocar no código. 

Caso tenha ficado muito confuso, a gente pode bater um papo melhor sobre!

File: extractors/main.py [hunk 1]
```diff
@@ -4,18 +4,27 @@
...
+        cnac = ExtractCNAC()
+        cnac.download()
+    elif source == "RCIV":
+        rciv = ExtractRCIV(
+            initial_date=datetime(2021, 1, 1), final_dalte=datetime(2021, 2, 28)
+        )
+        rciv.download()
+    elif source == "SRAG":
+        srag = ExtractSRAG()
+        srag.download()
```
Comment (davikawasaki) [inline extractors/main.py]: I'm down with @brbraziel suggestion. This way we can make it generic and avoid an overload of conditionals in here:

```suggestion
import json
import os
import importlib.util

def main(source_path, source_args):
    try:
       source_name = os.path.splitext(os.path.basename(source_path))[0]
       spec = importlib.util.spec_from_file_location(source_name, source_path)
       module = importlib.util.module_from_spec(spec)
       module_inst = module(*source_args)
       module_inst.download()
    except:
       print('dynamic import not successful')
```

Comment (davikawasaki) [reply extractors/main.py]: Just the idea here, I didn't test to see if this dynamic import works, especially with **multiple args**.

File: extractors/main.py [hunk 2]
```diff
@@ -4,18 +4,27 @@
...
 
-
-main()
+if __name__ == "__main__":
+    assert sys.argv[1].upper() in [
+        "CNAC",
+        "RCIV",
+        "SRAG",
+    ], "Invalid argument. Please make sure your source argument is either CNAC, RCIV or SRAG."
+    main(sys.argv[1].upper())
```
Comment (davikawasaki) [inline extractors/main.py]: JSON structure:

```json
{
    "CNAC": {
       "source_path": "caminho/do/arquivo/cnac.py",
       "source_args": null
    },
    "RCIV": {
       "source_path": "caminho/do/arquivo/rciv.py",
       "source_args": ["2021-01-01 00:00:00", "2021-02-28 00:00:00"]
    }
} 
```

```suggestion
    with open('path_to_file/dict.json') as f:
       dict = json.load(f)

    assert sys.argv[1].upper() in dict.keys(),
       "Invalid argument. Please make sure your source argument is either CNAC, RCIV or SRAG."
       
    dict_item = dict[sys.argv[1].upper()]
    main(dict_item['source_path'], dict_item['source_args'])
```

**Review by davikawasaki — CHANGES_REQUESTED** [review]
Well done. Just complementing @brbraziel suggestion with some recommended code. Bear in mind that I'm not a python developer, so there might be errors along the way :)

**Review by brunobraziel — APPROVED** [review]

**Review by davikawasaki — APPROVED** [review]

**Pull request merged**